application/comit_node/src/http_api/rfc003/routes.rs

```diff
@@ -0,0 +1,119 @@
+pub use crate::http_api::rfc003::handlers::{GetAction, GetActionQueryParams, PostAction};
+use crate::{
+    http_api::{
+        self,
+        problem::HttpApiProblemStdError,
+        rfc003::handlers::{
+            handle_get_action, handle_get_swap, handle_get_swaps, handle_post_action,
+            handle_post_swap, ActionName, GetSwapResource, SwapRequestBodyKind,
+        },
+    },
+    swap_protocols::{
+        rfc003::{alice::AliceSpawner, state_store::StateStore, SecretSource},
+        MetadataStore, SwapId,
+    },
+};
+use http_api_problem::HttpApiProblem;
+use hyper::header;
+use rustic_hal::HalResource;
+use std::sync::Arc;
+use warp::{Rejection, Reply};
+
+pub const PROTOCOL_NAME: &str = "rfc003";
+pub fn swap_path(id: SwapId) -> String {
+    format!("/{}/{}/{}", http_api::PATH, PROTOCOL_NAME, id)
+}
+
+#[allow(clippy::needless_pass_by_value)]
+pub fn post_swap<A: AliceSpawner>(
+    alice_spawner: A,
+    secret_source: Arc<dyn SecretSource>,
+    request_body_kind: SwapRequestBodyKind,
+) -> Result<impl Reply, Rejection> {
+    handle_post_swap(&alice_spawner, secret_source.as_ref(), request_body_kind)
+        .map(|swap_created| {
+            let body = warp::reply::json(&swap_created);
+            let response =
+                warp::reply::with_header(body, header::LOCATION, swap_path(swap_created.id));
+            warp::reply::with_status(response, warp::http::StatusCode::CREATED)
+        })
+        .map_err(|problem| warp::reject::custom(HttpApiProblemStdError::from(problem)))
```

Comment:
:x: Can you please unify all the router functions to use:

```rust
.map_err(into_rejection)
```

where `into_rejction` is:

```rust
fn into_rejection(problem: HttpApiProblem) -> Rejection {
    warp::reject::custom(HttpApiProblemStdError::from(problem))
}
```